### PR TITLE
Packages: Move `e2e-test-utils` to new structure

### LINKED ISSUE #6048: Packages: Move `e2e-test-utils` to new structure
## Feature Description

This package is a bit different from others as it just contains test utils, but would be good to separate these from the test themselves. Similar to how Gutenberg has e2e-tests and e2e-test-utils packages.

* [ ] Move code from `tests/e2e/utils` to `/packages/e2e-test-utils/src/`
* [ ] Update imports where necessary.  
  Use `@web-stories-wp/...` imports for anything that is already a package, relative imports for everything else.
* [ ] Create `/packages/e2e-test-utils/package.json` referencing its dependencies
* [ ] Create `/packages/e2e-test-utils/README.md` with short documentation.

## Additional Context

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

### PR DESCRIPTION
**Pull request by spacedmonkey**
## Summary

Move files into the new e2e-test-utils to new package structure. 

### QA

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

### UAT

- [ ] UAT should use the same steps as above.

This PR can be tested by following these steps:

1.

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testiing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #6048

### CODE REVIEW

**Comment by spacedmonkey** [comment]
@swissspidy It is worth noting, that `@wordpress/e2e-test-utils` is still installing heavily used in our end to end tests. I can import these in this new package and export them, but I am not sure I see the point. The e2e tests are heavily linked to WordPress so will not be decoupled, I believe. Willing to do this work, if you think it makes sense..

File: packages/e2e-test-utils/src/uploadMedia.js [hunk 1]
```diff
@@ -16,7 +16,7 @@
 /**
  * Internal dependencies
  */
-import { uploadFile } from './';
+import { uploadFile } from './index';
```
Comment (swissspidy) [inline packages/e2e-test-utils/src/uploadMedia.js]: can undo this change or just import from `visitDashboard.js` directly

File: packages/e2e-test-utils/src/setAnalyticsCode.js [hunk 2]
```diff
@@ -16,7 +16,7 @@
 /**
  * Internal dependencies
  */
-import { visitDashboard } from './';
+import { visitDashboard } from './index';
```
Comment (swissspidy) [inline packages/e2e-test-utils/src/setAnalyticsCode.js]: can undo this change or just import from `visitDashboard.js` directly

File: .eslintrc [hunk 3]
```diff
@@ -352,6 +352,14 @@
...
       }
     },
+    {
+      "files": ["packages/e2e-test-utils/src/*.js"],
+      "rules": {
+        "eslint-comments/no-unused-disable": "off",
+        "jest/no-hooks": "off",
+        "no-undef": "off"
+      }
+    },
```
Comment (swissspidy) [inline .eslintrc]: Why was this added?

**Comment by swissspidy** [comment]
> @swissspidy It is worth noting, that `@wordpress/e2e-test-utils` is still installing heavily used in our end to end tests. I can import these in this new package and export them, but I am not sure I see the point. The e2e tests are heavily linked to WordPress so will not be decoupled, I believe. Willing to do this work, if you think it makes sense..

yeah no I was not planning on doing that. can keep as-is.

File: .eslintrc (hunk 3 above)
Comment (spacedmonkey) [reply .eslintrc]: This linter didn't like page / expected.

File: packages/e2e-test-utils/src/setAnalyticsCode.js (hunk 2 above)
Comment (spacedmonkey) [reply packages/e2e-test-utils/src/setAnalyticsCode.js]: Sorry, my ide keeps this. https://github.com/google/web-stories-wp/pull/6275/commits/d8ba44562e453fd3a8f318f5d0028b8ea510bb41

File: .eslintrc (hunk 3 above)
Comment (swissspidy) [reply .eslintrc]: The correct way of fixing that would be to extend the `files` array here to include `packages/e2e-test-utils/src/**/*.js`:

https://github.com/google/web-stories-wp/blob/446bbd5f30075f8f74884a1f77894749a49b746c/.eslintrc#L333-L354

File: tests/e2e/specs/editor/media/insertMediaFromDialog.js [hunk 4]
```diff
@@ -13,20 +13,30 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-
+/* global __dirname */
```
Comment (swissspidy) [inline tests/e2e/specs/editor/media/insertMediaFromDialog.js]: Instead of declaring `global __dirname` everywhere, the correct way of fixing this would be to extend `files` here to include `packages/e2e-test-utils/src/**/*.js`:

https://github.com/google/web-stories-wp/blob/446bbd5f30075f8f74884a1f77894749a49b746c/.eslintrc#L362-L397

File: tests/e2e/specs/editor/media/insertMovVideo.js [hunk 5]
```diff
@@ -37,8 +38,13 @@ describe('Inserting .mov from dialog', () => {
...
     });
-
-    const fileName = await uploadFile('small-video.mov', false);
+    const testMediaPath = join(
+      __dirname,
+      '../../..',
+      'assets',
+      'small-video.mov'
+    );
+    const fileName = await uploadFile(testMediaPath, false);
```
Comment (swissspidy) [inline tests/e2e/specs/editor/media/insertMovVideo.js]: Having to use `join` and `__dirname` everytime to upload a file adds a lot of repetition. It was much more DRY before.

Comment (spacedmonkey) [reply tests/e2e/specs/editor/media/insertMovVideo.js]: To fix this, I would have to move the asset files into the e2e package, I am creating here. Should I do this?

Comment (spacedmonkey) [reply tests/e2e/specs/editor/media/insertMovVideo.js]: https://github.com/google/web-stories-wp/pull/6275/commits/6dc648c63ebf123a98300fe38bede9013caa014e

Comment (spacedmonkey) [reply tests/e2e/specs/editor/media/insertMovVideo.js]: Fixed with a work around.

Comment (swissspidy) [reply tests/e2e/specs/editor/media/insertMovVideo.js]: Oh, I was just about to reply to your comment, saying  that I now see why you did this and that it probably is fine (as long as the ESLint config is adequately updated).

It's an interesting workaround though 👀 I think we can use that for now

**Review by swissspidy — APPROVED** [review]
